### DISCUSSION TITLE
Proper Disposing of Timer in Caret Class

### DISCUSSION THREAD

**Discussion by itinerant-fox** [abbaye/WPFHexEditorControl#114] [Q&A] [answered]
https://github.com/abbaye/WpfHexEditorControl/blob/880b8baceb43cfa963f4c765f6bc5ccd0eb17175/Sources/WPFHexaEditor/Core/Caret.cs#L256

The above timer is not being correctly disposed. I think it needs to first unsubscribe the event handler and then dispose the Timer object. Instead it is just set as null. I noticed this because I keep getting TaskCancelled Exception on BlinkCaret() when exiting the application.
Do you think this should be so or am I missing something else?

**Comment by abbaye** [comment]
Yes I think you have reasons.

I can correct this or if you want put a pull request :)

**Comment by abbaye** [comment, marked as answer]
Corrected ;)
